### PR TITLE
SqlClient - Refactor SNIHandle Receive to use 'out' instead of 'ref'

### PR DESCRIPTION
**Pull request by kkurni**
https://github.com/dotnet/corefx/issues/4368

### CODE REVIEW

**Comment by kkurni** [comment]
CC @corivera @saurabh500

**Comment by stephentoub** [comment]
LGTM

**Comment by NickCraver** [comment]
+1

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs [hunk 1]
```diff
@@ -454,7 +455,10 @@ private uint ReportTcpSNIError(string errorMessage)
 
         private uint ReportErrorAndReleasePacket(SNIPacket packet, string errorMessage)
         {
-            packet.Release();
+            if (packet != null)
+            {
+                packet.Release();
```
Comment (davidfowl) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: Needs more c# 6:

```C#
packet?.Release();
```

Comment (NickCraver) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: I was going to comment here...but wasn't sure if the lib was using it at all yet, +1 if so.

Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: We're not using C# 6 yet in the repo.

Comment (NickCraver) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: In that case, +1 to doing *that* :) I'm curious what the blocker, if any, is...is there one?

Comment (davidfowl) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: :open_mouth:

Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: > I'm curious what the blocker, if any, is...is there one?

@weshaggard can comment.

Comment (weshaggard) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs]: We still have a number of build/test infrastructures that use the same code and not all of them are building with the Roslyn toolset yet. Hopefully we will be able to allow it in the coming months.

**Pull request merged**